frontend/src/libs/components/pagination/pagination.tsx

```diff
@@ -0,0 +1,150 @@
+
+  const handleChangePage = useCallback(
+    (page: number) => {
+      return () => {
+        onChangePage(page);
+      };
+    },
+    [onChangePage],
+  );
+
+  const visiblePages = useMemo(() => {
+    let pages;
+
+    switch (currentPage) {
+      case FIRST_PAGE: {
+        pages = [
+          currentPage,
+          currentPage + ONE_STEP_LENGTH,
+          currentPage + 2 * ONE_STEP_LENGTH,
+        ];
+        break;
+      }
+      case totalPages: {
+        pages = [
+          currentPage - 2 * ONE_STEP_LENGTH,
+          currentPage - ONE_STEP_LENGTH,
+          currentPage,
+        ];
+        break;
+      }
+      default: {
+        pages = [
+          currentPage - ONE_STEP_LENGTH,
+          currentPage,
+          currentPage + ONE_STEP_LENGTH,
+        ];
+      }
+    }
+
+    if (pages.length > totalPages) {
+      pages = pages.filter((page) => page >= FIRST_PAGE && page <= totalPages);
+    }
+
+    return pages;
+  }, [currentPage, totalPages]);
+
+  const isShowFirstPage = useMemo(() => {
+    return visiblePages[0] !== FIRST_PAGE;
+  }, [visiblePages]);
+  const isShowFirst3Dots = useMemo(() => {
+    return (
+      visiblePages[0] && visiblePages[0] >= FIRST_PAGE + 2 * ONE_STEP_LENGTH
+    );
+  }, [visiblePages]);
+
+  const isShowLastPage = useMemo(() => {
+    return visiblePages[visiblePages.length - 1] !== totalPages;
+  }, [visiblePages, totalPages]);
+  const isShowLast3Dots = useMemo(() => {
+    const lastVisiblePage = visiblePages[visiblePages.length - 1];
+
+    return (
+      lastVisiblePage && lastVisiblePage <= totalPages - 2 * ONE_STEP_LENGTH
+    );
+  }, [visiblePages, totalPages]);
+
+  return (
+    <div className={getValidClassNames(styles['wrapper'], className)}>
+      <IconButton
+        icon="arrowLeft"
+        label="On previous page"
+        onClick={handlePreviousPage}
+        className={styles['buttons']}
+        isDisabled={currentPage === FIRST_PAGE}
+      />
+
+      <div className={styles['item-wrapper']}>
+        {isShowFirstPage && (
+          <Button
+            label={FIRST_PAGE.toString()}
+            style="plain"
+            className={styles['item']}
+            onClick={handleChangePage(FIRST_PAGE)}
+          />
+        )}
+        {isShowFirst3Dots && <div>...</div>}
+        {visiblePages.map((page, index) => (
+          <Button
+            label={page.toString()}
+            style="plain"
+            key={index}
+            className={getValidClassNames(
+              styles['item'],
+              currentPage === page && styles['item-yellow'],
+            )}
+            onClick={handleChangePage(page)}
+          />
+        ))}
+        {isShowLast3Dots && <div>...</div>}
+        {isShowLastPage && (
+          <Button
+            label={totalPages.toString()}
+            style="plain"
+            className={styles['item']}
+            onClick={handleChangePage(totalPages)}
+          />
+        )}
+      </div>
```

Comment:
![image](https://user-images.githubusercontent.com/19575839/230774277-bf17be3d-e036-4aac-842b-4e23b8438be3.png)
Let's just have this approach of pagination components - just to simplify the implementation (but with ours current design)